### ISSUE TITLE
Deploy Weblate's components separately to allow for horizontal scaling

### ISSUE THREAD

**Issue by korewaChino** [WeblateOrg/helm#298]
### Describe the problem

Right now, the Weblate helm chart deploys itself as a single monolithic instance, meaning that every component runs in a single container. Causing slow startup times due to having to initialize every single component in the same container.

But according to https://docs.weblate.org/en/latest/admin/install/docker.html#scaling-horizontally, you can actually scale Weblate horizontally by deploying multiple instances of Weblate, with extra arguments to make them run a specific component.

Doing this allows for high-availability as individual components can be restarted without affecting the whole instance, meaning that when a certain component (i.e Celery, or the web server) fails, it can quickly restart and reconnect to the cluster without affecting downtime.

### Describe the solution you'd like

Extend the Helm chart to allow Weblate components to be deployed separately instead of running everything in a single binary. 

I got an example for a similar distributed Helm deployment that works similarly to what I'm describing, from Grafana Mimir which uses only 1 single Grafana Mimir container, but with extra arguments to target a specific component https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed

While that deployment has its own hash ring clustering, seems like our [distributed Docker compose](https://github.com/WeblateOrg/docker-compose/blob/main/docker-compose-split.yml) only relies on a common database backend, so this should be fine enough.

**Comment by nijel** [comment]
The containers need a shared storage as well (`/app/data` needs to be shared). NFS should work fine for that since Weblate 4.16.

**Comment by begemotik** [comment]
Hello. Would be nice to have, do you have any estimations?

**Comment by nijel** [comment]
The helm chart doesn't cover this, but https://github.com/WeblateOrg/helm/issues/54#issuecomment-4541503019 has some info how this can be setup.

**Comment by begemotik** [comment]
You are right that the chart supports running multiple replicas at the moment, but the original requests references to distributed setup from the documentation.
